--- qtmmlwidget.cpp.org	2007-01-15 22:34:37.000000000 +0100
+++ qtmmlwidget.cpp	2007-01-15 22:34:37.000000000 +0100
@@ -3970,9 +3970,6 @@
 void MmlNode::paint(QPainter *p)
 {
     p->save();
-    p->setViewport(deviceRect());
-    p->setWindow(myRect());
-
 
     QColor fg = color();
     QColor bg = background();
@@ -4219,7 +4216,9 @@
     p->save();
     p->setFont(fn);
 
-    p->drawText(0, fm.strikeOutPos(), m_text);
+    QPoint dPos = devicePoint(relOrigin());
+    p->drawText(dPos.x(), dPos.y() + fm.strikeOutPos(), m_text);
+
     p->restore();
 }
 
